Mathlib/NumberTheory/NumberField/Units/Basic.lean

```diff
@@ -135,8 +135,18 @@ instance : Nonempty (torsion K) := One.instNonempty
 /-- The torsion subgroup is cyclic. -/
 instance [NumberField K] : IsCyclic (torsion K) := subgroup_units_cyclic _
 
-/-- The order of the torsion subgroup as a positive integer. -/
-def torsionOrder [NumberField K] : ℕ+ := ⟨Fintype.card (torsion K), Fintype.card_pos⟩
+/-- The order of the torsion subgroup. -/
+def torsionOrder [NumberField K] : ℕ := Fintype.card (torsion K)
+
+instance [NumberField K] : NeZero (torsionOrder K) := by
+  unfold torsionOrder
+  infer_instance
```

Comment:
Can you use `inferInstanceAs` here?

Reply:
Oh, yes, I didn't think about that. Thanks!
There is still one last (coming soon) PR before #17914 is ready for review.